compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangStreamType.java

```diff
@@ -34,6 +34,8 @@
     public BLangType constraint;
     public BLangType error;
 
+    public boolean inferType;
```

Comment:
Shall we check if we can adopt an approach similar to how `future` is handled, instead of introducing this?

Reply:
Need to discuss with Chiran

Reply:
Looks like I have to come up with a default type which can handle any expected type.

Reply:
I think this is bit complex
```
 stream a = new stream<int, error?>();
 stream b = new stream<int>();
 ```
 
 We need to take decisions for completion type therefore we need some boolean/flag to tell us that we need to have a completion type based on RHS. I believe you need to avoid the boolan/flag?

Reply:
Had a offline chat with Maryam and resolved this. The way I understood is we need to infer but spec suggests `stream` should accept all possible stream types